sdk/nodejs/runtime/resource.ts

```diff
@@ -1177,6 +1177,9 @@ export function registerResourceOutputs(res: Resource, outputs: Inputs | Promise
                         (excessiveDebugOutput ? `, outputs=${JSON.stringify(outputsObj)}` : ``),
                 );
 
+                const store = getStore();
+                store.currentExportMap = resolved.outputs;
```

Comment:
We're writing this for every output registration? Doesn't that mean we could end up with any component resource being the last due to running inside an apply and so not actually getting the stack outputs here. I think this needs to be moved to the stack class or protected here at least by checking the type name.